clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1864,16 +1864,20 @@ static void printArguments(ASTContext &Ctx, raw_ostream &ArgOS,
                            const PrintingPolicy &P);
 
 static void printArgument(ASTContext &Ctx, raw_ostream &ArgOS,
-                          TemplateArgument Arg, const PrintingPolicy &P) {
+                          TemplateArgument Arg, const PrintingPolicy &P,
+                          bool FirstArg) {
   switch (Arg.getKind()) {
   case TemplateArgument::ArgKind::Pack: {
+    if (Arg.pack_size() && !FirstArg)
```

Comment:
Why is the comma being managed by this function?  It seems that whoever is calling the printArgument should make sure that the comma is appended correctly.

Reply:
Moved it back